bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/BridgeDiscoveryParticipant.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    @Override
+    public String getServiceType() {
+        return "_http._tcp.local.";
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo serviceInfo) {
+        logger.trace("ServiceInfo: {}", serviceInfo);
+
+        final ThingUID uid = getThingUID(serviceInfo);
+        if (uid == null) {
+            return null;
+        }
+
+        Map<String, Object> properties = new HashMap<>(1);
+        properties.put("ipAddress", uid.getId().replace('-', '.'));
+
+        return DiscoveryResultBuilder.create(uid).withLabel("Smart Home Controller").withProperties(properties)
+                .withTTL(60 * 60 * 24).build();
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(ServiceInfo serviceInfo) {
+        String ipAddress = getBridgeAddress(serviceInfo);
+        if (ipAddress.isEmpty()) {
+            return null;
+        }
+        return new ThingUID(BoschSHCBindingConstants.THING_TYPE_SHC, ipAddress.replace('.', '-'));
+    }
+
+    private String getBridgeAddress(ServiceInfo serviceInfo) {
+        InetAddress[] addresses = serviceInfo.getInetAddresses();
+        if (addresses.length == 0) {
+            return "";
+        }
+
+        // Try to get public SHC info from address
+        String ipAddress = addresses[0].getHostAddress();
+
+        SslContextFactory sslContextFactory = new SslContextFactory.Client.Client(true); // Accept all certificates
+        sslContextFactory.setTrustAll(true);
+        sslContextFactory.setValidateCerts(false);
+        sslContextFactory.setValidatePeerCerts(false);
+        sslContextFactory.setEndpointIdentificationAlgorithm(null);
+
+        HttpClient httpClient = new HttpClient(sslContextFactory);
+        String url = String.format("https://%s:8446/smarthome/public/information", ipAddress);
+        logger.trace("Discovering {}", url);
+        try {
+            httpClient.start();
+            ContentResponse contentResponse = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            // check HTTP status code
+            if (!HttpStatus.getCode(contentResponse.getStatus()).isSuccess()) {
+                logger.debug("Discovering failed with status code: {}", contentResponse.getStatus());
+                return "";
+            }
+
+            String content = contentResponse.getContentAsString();
+            logger.info("Discovered SHC at address {}, public info {}", ipAddress, content);
+            PublicInformation versionInfo = gson.fromJson(content, PublicInformation.class);
+            return versionInfo.shcIpAddress;
+
+        } catch (NullPointerException | InterruptedException | TimeoutException | ExecutionException e) {
+            logger.trace("Discovering failed with exception: {}", e);
+            return "";
+        } catch (Exception e) {
+            logger.trace("Discovering failed in http client start: {}", e);
+            return "";
+        }
+    }
+}
```

Comment:
I get two errors from Code Analysis as well:
```
[ERROR] org.openhab.binding.boschshc.internal.discovery.BridgeDiscoveryParticipant.java:[131]
Count of placeholder(1) is not equal to count of parameter(0)
[ERROR] org.openhab.binding.boschshc.internal.discovery.BridgeDiscoveryParticipant.java:[134]
Count of placeholder(1) is not equal to count of parameter(0)
```

Reply:
Plus a code format warning:
```
[WARNING] D:\Projects\openHAB\openhab-addons\bundles\org.openhab.binding.boschshc\src\main\java\org\openhab\binding\boschshc\internal\discovery\BridgeDiscoveryParticipant.java:[128,20] Potential null pointer access: The variable versionInfo may be null at this location
```

Reply:
Will fix the errors and warnings and format the code.

Reply:
> Any idea what could cause this? I used a fresh docker container with openHAB 3.0.1 ( openhab/openhab:3.0.1)

I use the same openHAB 3.0.1 version as an extracted zip and do not see the errors.
Perhaps you can try the same on your local machine, because I will need some effort to set up a docker environment.

Reply:
@GerdZanker Will try again and if it does not work I will try with the zipped version. I am really excited to discover my controller automatically :)

Reply:
@GerdZanker Strange, with the zipped version it worked :/ I will have to see if I get it to run again, but the discovery worked :) I had an idea to improve the discovery, but do not know if they are possible/feasible:
- Ask for the System Password when the bridge thing is added (maybe something like "asking for additional configuration parameters" is supported by openHAB)